### PR TITLE
(Deposit/Withdraw) Fix IBC status provider

### PR DESCRIPTION
**Pull request by jonator**
## What is the purpose of the change:

Fixed some issues with IBC status provider (it's tx query), as well as invalidating and preventing queries when txs are sent. It seems there's still some rare edge case bugs with balances for the selected asset being incorrect.

### Linear Task

[Linear Task URL](https://linear.app/osmosis/issue/FE-621/fix-ibc-status-provider)

## Brief Changelog

* Fix /txs/ endpoint in IBC status provider
* Fix deserializing tx events in IBC status provider (some nodes return the tx.event array base64 encoded, in which case the raw_log will be present with decoded events array when parsed as JSON. If raw_log is not present, the tx.event array should be decoded and used)
* Invalidate user account queries used by bridge flow when a tx is confirmed
* Prevent subtracting into a negative value when balance is less than the gas fee. Set to 0 instead.

## Testing and Verifying

Transfers should work and be reflected in transfer history table and toasts.

### CODE REVIEW

File: packages/web/components/bridge/immersive/amount-and-review-screen.tsx [hunk 1]
```diff
@@ -103,8 +105,12 @@ export const AmountAndReviewScreen = observer(
           ? fromAsset?.supportedVariants[toAsset?.address ?? ""]
           : toAsset?.supportedVariants[fromAsset?.address ?? ""],
       onTransfer: () => {
+        setToAsset(undefined);
+        setFromAsset(undefined);
         setCryptoAmount("0");
         setFiatAmount("0");
+
+        refetchUserQueries(apiUtils);
```
Comment (jonator) [inline packages/web/components/bridge/immersive/amount-and-review-screen.tsx:113]: Makes sure balances are accurate when restarting the d/w flow

File: packages/web/components/bridge/immersive/amount-and-review-screen.tsx [hunk 2]
```diff
@@ -103,8 +105,12 @@ export const AmountAndReviewScreen = observer(
           ? fromAsset?.supportedVariants[toAsset?.address ?? ""]
           : toAsset?.supportedVariants[fromAsset?.address ?? ""],
       onTransfer: () => {
+        setToAsset(undefined);
+        setFromAsset(undefined);
```
Comment (jonator) [inline packages/web/components/bridge/immersive/amount-and-review-screen.tsx:109]: Resets state when restarting the flow

File: packages/web/components/bridge/immersive/use-bridge-quotes.ts [hunk 3]
```diff
@@ -160,6 +167,7 @@ export const useBridgeQuotes = ({
         },
         {
           enabled:
+            !isTxPending &&
```
Comment (jonator) [inline packages/web/components/bridge/immersive/use-bridge-quotes.ts:170]: Prevents queries from running while the tx is confirming/just confirmed

File: packages/web/components/bridge/immersive/use-bridge-quotes.ts [hunk 4]
```diff
@@ -377,28 +385,22 @@ export const useBridgeQuotes = ({
           `${providerId}${JSON.stringify(params)}`,
           inputCoin.trim(true).toString(),
           isWithdraw,
-          toAddress ?? "" // use osmosis account (destinationAddress) for account keys (vs any EVM account)
+          (isWithdraw ? fromAddress : toAddress) ?? "" // use osmosis account (destinationAddress) for account keys (vs any EVM account)
```
Comment (jonator) [inline packages/web/components/bridge/immersive/use-bridge-quotes.ts:388]: Transfer history is keyed and filtered by Osmosis account that is currently connected

File: packages/web/stores/index.tsx [hunk 5]
```diff
@@ -26,6 +26,8 @@ function refetchUserQueries(apiUtils: ReturnType<typeof api.useUtils>) {
   apiUtils.edge.pools.getUserSharePool.invalidate();
   apiUtils.edge.pools.getSharePoolBondDurations.invalidate();
   apiUtils.local.balances.getUserBalances.invalidate();
+  apiUtils.local.bridgeTransfer.getSupportedAssetsBalances.invalidate();
+  apiUtils.edge.assets.getImmersiveBridgeAssets.invalidate();
```
Comment (jonator) [inline packages/web/stores/index.tsx:30]: Added these to ensure old balance data is not displayed in the d/w flow after a transfer is started / completed. @JoseRFelix LMK if I'm missing any

File: yarn.lock
Comment (jonator) [inline yarn.lock:1]: Not sure why yarn lock was changed when installing deps on stage

File: packages/web/components/bridge/immersive/amount-and-review-screen.tsx (hunk 1 above)
Comment (JoseRFelix) [reply packages/web/components/bridge/immersive/amount-and-review-screen.tsx:113]: Nit: I believe we should add this same GH comment as a code comment for this line

File: packages/web/components/bridge/immersive/use-bridge-quotes.ts (hunk 3 above)
Comment (JoseRFelix) [reply packages/web/components/bridge/immersive/use-bridge-quotes.ts:170]: Let's add this comment here too

**Review by JoseRFelix — APPROVED** [review]
LGTM!

**Pull request merged**